File: modules/app/src/main/java/org/artembogomolova/demo/webapp/domain/core/Person.java
package org.artembogomolova.demo.webapp.domain.core;

import java.util.Date;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.artembogomolova.demo.webapp.domain.IdentifiedEntity;
import org.artembogomolova.demo.webapp.domain.auth.User;

@Entity
@Table(name = "persons")
@Getter
@Setter
@ToString(exclude = {"estateAddress"})
@EqualsAndHashCode
public class Person extends IdentifiedEntity {

  private static final Long serialVersionUID = 1L;

  private String name;
  private String surname;
  private String patronymic;
  @Temporal(TemporalType.TIMESTAMP)
  private Date birthDate;
  private String phone;
  @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REMOVE, CascadeType.DETACH})
  @JoinColumn(name = "estate_address_id", columnDefinition = "bigint")
  private PhysicalAddress estateAddress;
  @OneToOne
  @JoinColumn(name = "id")
  private User user;

  public Date getBirthDate() {
    if (birthDate == null) {
      return null;
    }
    return new Date(birthDate.getTime());
  }

  public void setBirthDate(Date birthDate) {
    this.birthDate = new Date(birthDate.getTime());
  }
}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/domain/core/Person.kt
package org.artembogomolova.demo.webapp.main.domain.core

import org.artembogomolova.demo.webapp.main.domain.IdentifiedEntity
import org.artembogomolova.demo.webapp.main.domain.auth.User
import java.util.Date
import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.OneToOne
import javax.persistence.Table
import javax.persistence.Temporal
import javax.persistence.TemporalType


@Entity
@Table(name = "persons")
class Person(
    @Column(nullable = false)
    val name: String,
    @Column(nullable = false)
    val surname: String,
    @Column(nullable = false)
    val patronymic: String,
    @Column(nullable = false)
    val phone: String,
    @ManyToOne(cascade = [CascadeType.ALL])
    @JoinColumn(name = "estate_address_id", columnDefinition = "bigint", nullable = false)
    var estateAddress: PhysicalAddress?,
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "birthDate", nullable = false)
    private var _birthDate: Date? = null,
    @OneToOne(mappedBy = "person")
    val user: User? = null
) : IdentifiedEntity() {

    var birthDate: Date?
        get() = if (_birthDate != null) {
            Date(_birthDate!!.time)
        } else {
            null
        }
        set(value) {
            if (value != null) {
                _birthDate = Date(value.time)
            }
        }

    companion object {
        private const val serialVersionUID = 1L
    }
}